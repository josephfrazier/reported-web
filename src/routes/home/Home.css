/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

@import url('../../components/variables.css');

* {
  box-sizing: border-box;
}

.root {
  padding-left: 20px;
  padding-right: 20px;
}

.container {
  margin: 1em auto;
  padding: 0 0 40px;
  max-width: var(--max-content-width);
}

summary {
  outline: none;
}

label {
  display: block;
  margin: 1em 0;
}

input[type='search'] {
  appearance: searchfield;
}

input[type='search']::-webkit-search-cancel-button {
  appearance: searchfield-cancel-button;
}
