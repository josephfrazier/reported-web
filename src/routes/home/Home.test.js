/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import '@babel/polyfill';
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../components/App';
import Home from './Home';

require('timezone-mock').register('US/Eastern');
require('jest-mock-now')();

describe('Home', () => {
  test('renders children correctly', () => {
    const wrapper = renderer
      .create(
        <App context={{ insertCss: () => {}, fetch: () => {}, pathname: '' }}>
          <Home />
        </App>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
