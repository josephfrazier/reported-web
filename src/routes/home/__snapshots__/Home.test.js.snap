// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders children correctly 1`] = `
<div
  aria-disabled={false}
  className="root"
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "overflowY": "scroll",
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    className="container"
  >
    <main>
      <h1>
        <a
          href="https://reportedly.weebly.com/"
          style={
            Object {
              "textDecoration": "underline",
            }
          }
        >
          Reported
        </a>
      </h1>
      <details
        onToggle={[Function]}
        open={true}
      >
        <summary>
          Create/Edit User (click to expand)
        </summary>
        <form
          onSubmit={[Function]}
        >
          <fieldset
            disabled={false}
          >
            <label>
              Email:
               
              <input
                autoComplete="email"
                name="email"
                onChange={[Function]}
                onInvalid={[Function]}
                required={true}
                type="email"
                value=""
              />
            </label>
            <label>
              Password (this is saved on your device, so use a password you don't use anywhere else): 
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <input
                  autoComplete="current-password"
                  name="password"
                  onChange={[Function]}
                  onInvalid={[Function]}
                  required={true}
                  type="password"
                  value=""
                />
                 
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Show
                </button>
                 
                <button
                  onClick={[Function]}
                  type="button"
                >
                  Reset
                </button>
              </div>
            </label>
            <button
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Sign Up / Log In
            </button>
            <label>
              First Name:
               
              <input
                autoComplete="given-name"
                name="FirstName"
                onChange={[Function]}
                onInvalid={[Function]}
                required={true}
                type="text"
                value=""
              />
            </label>
            <label>
              Last Name:
               
              <input
                autoComplete="family-name"
                name="LastName"
                onChange={[Function]}
                onInvalid={[Function]}
                required={true}
                type="text"
                value=""
              />
            </label>
            <label>
              Building Number:
               
              <input
                name="Building"
                onChange={[Function]}
                onInvalid={[Function]}
                required={true}
                type="text"
                value=""
              />
            </label>
            <label>
              Street Name:
               
              <input
                name="StreetName"
                onChange={[Function]}
                onInvalid={[Function]}
                required={true}
                type="text"
                value=""
              />
            </label>
            <label>
              Apartment Number:
               
              <input
                name="Apt"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
            <label>
              Borough:
               
              <select
                name="Borough"
                onChange={[Function]}
                value="Bronx"
              >
                <option
                  value="Bronx"
                >
                  Bronx
                </option>
                <option
                  value="Brooklyn"
                >
                  Brooklyn
                </option>
                <option
                  value="Manhattan"
                >
                  Manhattan
                </option>
                <option
                  value="Queens"
                >
                  Queens
                </option>
                <option
                  value="Staten Island"
                >
                  Staten Island
                </option>
                <option
                  value="NOT WITHIN NEW YORK CITY"
                >
                  NOT WITHIN NEW YORK CITY
                </option>
              </select>
            </label>
            <label>
              Phone Number:
               
              <input
                autoComplete="tel"
                name="Phone"
                onChange={[Function]}
                onInvalid={[Function]}
                required={true}
                type="tel"
                value=""
              />
            </label>
            <label>
              <input
                checked={false}
                name="testify"
                onChange={[Function]}
                type="checkbox"
              />
               
              I'm willing to testify at a hearing, which can be done by phone.
            </label>
            <button
              disabled={false}
              type="submit"
            >
              Save
            </button>
          </fieldset>
        </form>
      </details>
      <form
        onSubmit={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
      >
        <fieldset
          disabled={false}
        >
          <div
            className="_react-file-reader-input"
            onClick={[Function]}
            style={
              Object {
                "float": "left",
                "margin": "1px",
              }
            }
          >
            <input
              accept="image/*"
              multiple={true}
              onChange={[Function]}
              onClick={[Function]}
              style={
                Object {
                  "position": "absolute",
                  "top": "-9999px",
                }
              }
              type="file"
            />
            <button
              type="button"
            >
              Add picture(s)
            </button>
          </div>
          <div
            className="_react-file-reader-input"
            onClick={[Function]}
            style={
              Object {
                "float": "left",
                "margin": "1px",
              }
            }
          >
            <input
              accept="video/*"
              multiple={true}
              onChange={[Function]}
              onClick={[Function]}
              style={
                Object {
                  "position": "absolute",
                  "top": "-9999px",
                }
              }
              type="file"
            />
            <button
              type="button"
            >
              Add video(s)
            </button>
          </div>
          <div
            style={
              Object {
                "clear": "both",
                "display": "flex",
                "flexWrap": "wrap",
              }
            }
          />
          <label>
            License/Medallion:
            <input
              list="plateSuggestion"
              onChange={[Function]}
              placeholder=""
              required={true}
              type="search"
              value=""
            />
            <datalist
              id="plateSuggestion"
            >
              
            </datalist>
            <select
              name="licenseState"
              onChange={[Function]}
              style={
                Object {
                  "marginTop": "0.5rem",
                }
              }
              value="NY"
            >
              <option
                value="AL"
              >
                Alabama
              </option>
              <option
                value="AK"
              >
                Alaska
              </option>
              <option
                value="AZ"
              >
                Arizona
              </option>
              <option
                value="AR"
              >
                Arkansas
              </option>
              <option
                value="CA"
              >
                California
              </option>
              <option
                value="CO"
              >
                Colorado
              </option>
              <option
                value="CT"
              >
                Connecticut
              </option>
              <option
                value="DE"
              >
                Delaware
              </option>
              <option
                value="FL"
              >
                Florida
              </option>
              <option
                value="GA"
              >
                Georgia
              </option>
              <option
                value="HI"
              >
                Hawaii
              </option>
              <option
                value="ID"
              >
                Idaho
              </option>
              <option
                value="IL"
              >
                Illinois
              </option>
              <option
                value="IN"
              >
                Indiana
              </option>
              <option
                value="IA"
              >
                Iowa
              </option>
              <option
                value="KS"
              >
                Kansas
              </option>
              <option
                value="KY"
              >
                Kentucky
              </option>
              <option
                value="LA"
              >
                Louisiana
              </option>
              <option
                value="ME"
              >
                Maine
              </option>
              <option
                value="MD"
              >
                Maryland
              </option>
              <option
                value="MA"
              >
                Massachusetts
              </option>
              <option
                value="MI"
              >
                Michigan
              </option>
              <option
                value="MN"
              >
                Minnesota
              </option>
              <option
                value="MS"
              >
                Mississippi
              </option>
              <option
                value="MO"
              >
                Missouri
              </option>
              <option
                value="MT"
              >
                Montana
              </option>
              <option
                value="NE"
              >
                Nebraska
              </option>
              <option
                value="NV"
              >
                Nevada
              </option>
              <option
                value="NH"
              >
                New Hampshire
              </option>
              <option
                value="NJ"
              >
                New Jersey
              </option>
              <option
                value="NM"
              >
                New Mexico
              </option>
              <option
                value="NY"
              >
                New York
              </option>
              <option
                value="NC"
              >
                North Carolina
              </option>
              <option
                value="ND"
              >
                North Dakota
              </option>
              <option
                value="OH"
              >
                Ohio
              </option>
              <option
                value="OK"
              >
                Oklahoma
              </option>
              <option
                value="OR"
              >
                Oregon
              </option>
              <option
                value="PA"
              >
                Pennsylvania
              </option>
              <option
                value="RI"
              >
                Rhode Island
              </option>
              <option
                value="SC"
              >
                South Carolina
              </option>
              <option
                value="SD"
              >
                South Dakota
              </option>
              <option
                value="TN"
              >
                Tennessee
              </option>
              <option
                value="TX"
              >
                Texas
              </option>
              <option
                value="UT"
              >
                Utah
              </option>
              <option
                value="VT"
              >
                Vermont
              </option>
              <option
                value="VA"
              >
                Virginia
              </option>
              <option
                value="WA"
              >
                Washington
              </option>
              <option
                value="WV"
              >
                West Virginia
              </option>
              <option
                value="WI"
              >
                Wisconsin
              </option>
              <option
                value="WY"
              >
                Wyoming
              </option>
            </select>
            <br />
          </label>
          <label>
            I was:
             
            <select
              name="typeofuser"
              onChange={[Function]}
              value="Cyclist"
            >
              <option
                value="Cyclist"
              >
                Cyclist
              </option>
              <option
                value="Walker"
              >
                Walker
              </option>
              <option
                value="Passenger"
              >
                Passenger
              </option>
            </select>
          </label>
          <label>
            Type:
             
            <select
              name="typeofcomplaint"
              onChange={[Function]}
              value="Blocked the bike lane"
            >
              <option
                value="Blocked the bike lane"
              >
                Blocked the bike lane
              </option>
              <option
                value="Blocked the crosswalk"
              >
                Blocked the crosswalk
              </option>
              <option
                value="Honked horn (no emergency)"
              >
                Honked horn (no emergency)
              </option>
              <option
                value="Failed to yield to pedestrian"
              >
                Failed to yield to pedestrian
              </option>
              <option
                value="Drove aggressively"
              >
                Drove aggressively
              </option>
              <option
                value="Was on a cell phone while driving"
              >
                Was on a cell phone while driving
              </option>
              <option
                value="Refused to pick me up"
              >
                Refused to pick me up
              </option>
              <option
                value="Was courteous, kind or polite"
              >
                Was courteous, kind or polite
              </option>
              <option
                value="Went above and beyond to help"
              >
                Went above and beyond to help
              </option>
            </select>
          </label>
          <label>
            Where:
            <br />
            <button
              onClick={[Function]}
              style={
                Object {
                  "width": "100%",
                }
              }
              type="button"
            >
              
            </button>
          </label>
          <label>
            When:
             
            <input
              name="CreateDate"
              onChange={[Function]}
              required={true}
              type="datetime-local"
              value="2016-11-17T19:00"
            />
          </label>
          <label>
            Description:
             
            <textarea
              name="reportDescription"
              onChange={[Function]}
              value=""
            />
          </label>
          <label>
            <input
              checked={false}
              name="can_be_shared_publicly"
              onChange={[Function]}
              type="checkbox"
            />
             
            Allow the photos/videos, description, category, and location to be publicly displayed
          </label>
          <button
            disabled={false}
            style={
              Object {
                "width": "100%",
              }
            }
            type="submit"
          >
            Submit
          </button>
        </fieldset>
      </form>
      <br />
      <details
        onToggle={[Function]}
      >
        <summary>
          Previous Submissions
        </summary>
        <ul>
          Loading submissions...
        </ul>
      </details>
      <div
        style={
          Object {
            "float": "right",
          }
        }
      >
        <a
          aria-label="github"
          className="social-icon"
          href="https://github.com/josephfrazier/Reported-Web"
          rel="noopener"
          style={
            Object {
              "display": "inline-block",
              "height": "50px",
              "overflow": "hidden",
              "position": "relative",
              "verticalAlign": "middle",
              "width": "50px",
            }
          }
          target="_blank"
        >
          <div
            className="social-container"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          >
            <svg
              className="social-svg"
              style={
                Object {
                  "borderRadius": "50%",
                  "fillRule": "evenodd",
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
              viewBox="0 0 64 64"
            >
              <g
                className="social-svg-background"
                style={
                  Object {
                    "MozTransition": "fill 170ms ease-in-out",
                    "OTransition": "fill 170ms ease-in-out",
                    "WebkitTransition": "fill 170ms ease-in-out",
                    "fill": "transparent",
                    "msTransition": "fill 170ms ease-in-out",
                    "transition": "fill 170ms ease-in-out",
                  }
                }
              >
                <circle
                  cx="32"
                  cy="32"
                  r="31"
                />
              </g>
              <g
                className="social-svg-icon"
                style={
                  Object {
                    "MozTransition": "fill 170ms ease-in-out",
                    "OTransition": "fill 170ms ease-in-out",
                    "WebkitTransition": "fill 170ms ease-in-out",
                    "fill": "transparent",
                    "msTransition": "fill 170ms ease-in-out",
                    "transition": "fill 170ms ease-in-out",
                  }
                }
              >
                <path
                  d="M32,16c-8.8,0-16,7.2-16,16c0,7.1,4.6,13.1,10.9,15.2 c0.8,0.1,1.1-0.3,1.1-0.8c0-0.4,0-1.4,0-2.7c-4.5,1-5.4-2.1-5.4-2.1c-0.7-1.8-1.8-2.3-1.8-2.3c-1.5-1,0.1-1,0.1-1 c1.6,0.1,2.5,1.6,2.5,1.6c1.4,2.4,3.7,1.7,4.7,1.3c0.1-1,0.6-1.7,1-2.1c-3.6-0.4-7.3-1.8-7.3-7.9c0-1.7,0.6-3.2,1.6-4.3 c-0.2-0.4-0.7-2,0.2-4.2c0,0,1.3-0.4,4.4,1.6c1.3-0.4,2.6-0.5,4-0.5c1.4,0,2.7,0.2,4,0.5c3.1-2.1,4.4-1.6,4.4-1.6 c0.9,2.2,0.3,3.8,0.2,4.2c1,1.1,1.6,2.5,1.6,4.3c0,6.1-3.7,7.5-7.3,7.9c0.6,0.5,1.1,1.5,1.1,3c0,2.1,0,3.9,0,4.4 c0,0.4,0.3,0.9,1.1,0.8C43.4,45.1,48,39.1,48,32C48,23.2,40.8,16,32,16z"
                />
              </g>
              <g
                className="social-svg-mask"
                style={
                  Object {
                    "MozTransition": "fill 170ms ease-in-out",
                    "OTransition": "fill 170ms ease-in-out",
                    "WebkitTransition": "fill 170ms ease-in-out",
                    "fill": "black",
                    "msTransition": "fill 170ms ease-in-out",
                    "transition": "fill 170ms ease-in-out",
                  }
                }
              >
                <path
                  d="M0,0v64h64V0H0z M37.1,47.2c-0.8,0.2-1.1-0.3-1.1-0.8c0-0.5,0-2.3,0-4.4c0-1.5-0.5-2.5-1.1-3 c3.6-0.4,7.3-1.7,7.3-7.9c0-1.7-0.6-3.2-1.6-4.3c0.2-0.4,0.7-2-0.2-4.2c0,0-1.3-0.4-4.4,1.6c-1.3-0.4-2.6-0.5-4-0.5 c-1.4,0-2.7,0.2-4,0.5c-3.1-2.1-4.4-1.6-4.4-1.6c-0.9,2.2-0.3,3.8-0.2,4.2c-1,1.1-1.6,2.5-1.6,4.3c0,6.1,3.7,7.5,7.3,7.9 c-0.5,0.4-0.9,1.1-1,2.1c-0.9,0.4-3.2,1.1-4.7-1.3c0,0-0.8-1.5-2.5-1.6c0,0-1.6,0-0.1,1c0,0,1,0.5,1.8,2.3c0,0,0.9,3.1,5.4,2.1 c0,1.3,0,2.3,0,2.7c0,0.4-0.3,0.9-1.1,0.8C20.6,45.1,16,39.1,16,32c0-8.8,7.2-16,16-16c8.8,0,16,7.2,16,16 C48,39.1,43.4,45.1,37.1,47.2z"
                />
              </g>
            </svg>
          </div>
        </a>
         
        <a
          aria-label="twitter"
          className="social-icon"
          href="https://twitter.com/Reported_NYC"
          rel="noopener"
          style={
            Object {
              "display": "inline-block",
              "height": "50px",
              "overflow": "hidden",
              "position": "relative",
              "verticalAlign": "middle",
              "width": "50px",
            }
          }
          target="_blank"
        >
          <div
            className="social-container"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          >
            <svg
              className="social-svg"
              style={
                Object {
                  "borderRadius": "50%",
                  "fillRule": "evenodd",
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
              viewBox="0 0 64 64"
            >
              <g
                className="social-svg-background"
                style={
                  Object {
                    "MozTransition": "fill 170ms ease-in-out",
                    "OTransition": "fill 170ms ease-in-out",
                    "WebkitTransition": "fill 170ms ease-in-out",
                    "fill": "transparent",
                    "msTransition": "fill 170ms ease-in-out",
                    "transition": "fill 170ms ease-in-out",
                  }
                }
              >
                <circle
                  cx="32"
                  cy="32"
                  r="31"
                />
              </g>
              <g
                className="social-svg-icon"
                style={
                  Object {
                    "MozTransition": "fill 170ms ease-in-out",
                    "OTransition": "fill 170ms ease-in-out",
                    "WebkitTransition": "fill 170ms ease-in-out",
                    "fill": "transparent",
                    "msTransition": "fill 170ms ease-in-out",
                    "transition": "fill 170ms ease-in-out",
                  }
                }
              >
                <path
                  d="M48,22.1c-1.2,0.5-2.4,0.9-3.8,1c1.4-0.8,2.4-2.1,2.9-3.6c-1.3,0.8-2.7,1.3-4.2,1.6 C41.7,19.8,40,19,38.2,19c-3.6,0-6.6,2.9-6.6,6.6c0,0.5,0.1,1,0.2,1.5c-5.5-0.3-10.3-2.9-13.5-6.9c-0.6,1-0.9,2.1-0.9,3.3 c0,2.3,1.2,4.3,2.9,5.5c-1.1,0-2.1-0.3-3-0.8c0,0,0,0.1,0,0.1c0,3.2,2.3,5.8,5.3,6.4c-0.6,0.1-1.1,0.2-1.7,0.2c-0.4,0-0.8,0-1.2-0.1 c0.8,2.6,3.3,4.5,6.1,4.6c-2.2,1.8-5.1,2.8-8.2,2.8c-0.5,0-1.1,0-1.6-0.1c2.9,1.9,6.4,2.9,10.1,2.9c12.1,0,18.7-10,18.7-18.7 c0-0.3,0-0.6,0-0.8C46,24.5,47.1,23.4,48,22.1z"
                />
              </g>
              <g
                className="social-svg-mask"
                style={
                  Object {
                    "MozTransition": "fill 170ms ease-in-out",
                    "OTransition": "fill 170ms ease-in-out",
                    "WebkitTransition": "fill 170ms ease-in-out",
                    "fill": "#00aced",
                    "msTransition": "fill 170ms ease-in-out",
                    "transition": "fill 170ms ease-in-out",
                  }
                }
              >
                <path
                  d="M0,0v64h64V0H0z M44.7,25.5c0,0.3,0,0.6,0,0.8C44.7,35,38.1,45,26.1,45c-3.7,0-7.2-1.1-10.1-2.9 c0.5,0.1,1,0.1,1.6,0.1c3.1,0,5.9-1,8.2-2.8c-2.9-0.1-5.3-2-6.1-4.6c0.4,0.1,0.8,0.1,1.2,0.1c0.6,0,1.2-0.1,1.7-0.2 c-3-0.6-5.3-3.3-5.3-6.4c0,0,0-0.1,0-0.1c0.9,0.5,1.9,0.8,3,0.8c-1.8-1.2-2.9-3.2-2.9-5.5c0-1.2,0.3-2.3,0.9-3.3 c3.2,4,8.1,6.6,13.5,6.9c-0.1-0.5-0.2-1-0.2-1.5c0-3.6,2.9-6.6,6.6-6.6c1.9,0,3.6,0.8,4.8,2.1c1.5-0.3,2.9-0.8,4.2-1.6 c-0.5,1.5-1.5,2.8-2.9,3.6c1.3-0.2,2.6-0.5,3.8-1C47.1,23.4,46,24.5,44.7,25.5z"
                />
              </g>
            </svg>
          </div>
        </a>
         
        <a
          href="/electricitibikes"
          style={
            Object {
              "background": "black",
              "border": "1em solid black",
              "borderRadius": "2em",
              "textDecoration": "none",
            }
          }
        >
          <span
            aria-label="high voltage"
            role="img"
          >
            ⚡
          </span>
        </a>
      </div>
    </main>
  </div>
  <input
    accept="image/*, video/*"
    autoComplete="off"
    disabled={false}
    multiple={true}
    onChange={[Function]}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "opacity": 0.00001,
        "pointerEvents": "none",
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    type="file"
  />
</div>
`;
